courses/views.py

```diff
@@ -339,18 +339,50 @@ def dispatch(self, *args, **kwargs):
 
 @login_required
 def list_of_courses_to_add(request):
-    qs = Course.objects.all()
-    student = get_object_or_404(Student, name=request.user)
-    return render(
-        request,
-        'course_list_to_add.html',
-        {
-            'courses': qs,
-            'student_id': student.pk
-        }
-    )
+    if not request.user.is_instructor():
+        qs = Course.objects.all()
+        student = get_object_or_404(Student, name=request.user)
```

Comment:
you don't have to fetch it this way

simply using:
    request.user.student

will work